plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -189,46 +207,56 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
-                        return Optional.empty();
-                    }
-                    return Optional.of(columnMappingPushdown(
-                            StandardColumnMappings.decimalColumnMapping(DecimalType.createDecimalType(precision, Math.max(decimalDigits, 0)), RoundingMode.UNNECESSARY)));
-                })
-                .buildOrThrow();
-
-        ColumnMappingFunction columnMappingFunction = snowflakeColumnMappings.get(jdbcTypeName);
-        if (columnMappingFunction != null) {
-            return columnMappingFunction.convert(typeHandle);
+        switch (type) {
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+            case Types.TINYINT:
+                return Optional.of(tinyintColumnMapping());
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.REAL:
+                return Optional.of(realColumnMapping());
+            case Types.FLOAT:
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.NUMERIC:
+            case Types.DECIMAL: {
+                int precision = typeHandle.getRequiredColumnSize();
+                int scale = typeHandle.getRequiredDecimalDigits();
+                DecimalType decimalType = createDecimalType(precision, scale);
+                return Optional.of(decimalColumnMapping(decimalType, UNNECESSARY));
+            }
+            case Types.VARCHAR:
+                if (jdbcTypeName.equals("varchar")) {
+                    return Optional.of(varcharColumnMapping(typeHandle.getRequiredColumnSize(), typeHandle.getCaseSensitivity()));
+                }
+                // Some other Snowflake types (ARRAY, VARIANT, GEOMETRY, etc.) are also mapped to Types.VARCHAR, but
+                // they're unsupported.
+                return Optional.empty();
+            case Types.BINARY:
+                // Multiple Snowflake types are mapped into Types.BINARY
+                if (jdbcTypeName.equals("binary")) {
+                    return Optional.of(varbinaryColumnMapping());
+                }
+                // Some other Snowflake types (GEOMETRY in some cases, etc.) are also mapped to Types.BINARY, but
+                // they're unsupported.
+                return Optional.empty();
+            case Types.VARBINARY:
+                return Optional.of(varbinaryColumnMapping());
+            case Types.LONGVARBINARY:
+                return Optional.of(varbinaryColumnMapping());
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(DateType.DATE, (resultSet, columnIndex) -> LocalDate.ofEpochDay(resultSet.getLong(columnIndex)).toEpochDay(), snowFlakeDateWriter()));
```

Comment:
LocalDate looks redundant. I think we can simply use `ResultSet::getLong` instead.

Also, `snowFlakeDateWriter` should be renamed to `snowFlakeDateWrriteFunction`

Reply:
Yes. Actually I made the same change in https://github.com/trinodb/trino/pull/21012/files#diff-29a16717b13dd57493cadfa2a3548b812c3557b327827345f63c6d2bef6b4bdeR224. Can move it to this PR as well.